packages/backend/src/routing/v1/ai-sdk.ts

```diff
@@ -0,0 +1,141 @@
+import { Hono } from "hono";
+import { generateText, LanguageModel } from "ai";
+import superjson from "superjson";
+import { selectProvider } from "../selector.js";
+import { ProviderFactory } from "../../providers/factory.js";
+import { logger } from "../../utils/logger.js";
+import { LanguageModelV2Prompt } from "@ai-sdk/provider";
+
+// Interface for the deserialized AI SDK generateText request
+// This should match the parameters accepted by generateText()
+interface GenerateTextRequest {
```

Comment:
There's no point in defining this!   The type needed is already in the ai SDK library!